ADAL/src/ADAuthenticationErrorMap.m

```diff
@@ -102,7 +102,7 @@ + (void)initialize
             @(MSIDErrorServerInvalidRequest): @(AD_ERROR_SERVER_OAUTH),
             @(MSIDErrorServerInvalidClient): @(AD_ERROR_SERVER_OAUTH),
             @(MSIDErrorServerInvalidGrant): @(AD_ERROR_SERVER_OAUTH),
-            @(MSIDErrorServerInvalidScope): @(AD_ERROR_SERVER_OAUTH),
+            @(MSIDErrorServerInvalidScope): @(AD_ERROR_SERVER_AUTHORIZATION_CODE),
```

Comment:
I think this also applies to other oauth errors when we don't get authorization code. ADAL previously was always returning AD_ERROR_SERVER_AUTHORIZATION_CODE, when it gets error instead of authorization code. Now this maps it to AD_ERROR_SERVER_OAUTH. Do you mind updating other cases as well and ensuring that when we don't get authorization code successfully, we return correct error?